src/Codegen/Generator.hack

```diff
@@ -302,31 +305,23 @@ final class Generator {
             $objects[] = new EnumBuilder($enum_type, $enum->getName());
         }
 
-        // TODO: throw an error if no query fields?
-
-        if (!C\is_empty($query_fields)) {
-            $top_level_objects = vec[
-                new ObjectBuilder(
-                    new \Slack\GraphQL\__Private\CompositeType('Query', 'Query'),
-                    'Slack\\GraphQL\\Root',
-                    vec(Dict\sort_by_key($query_fields)),
-                    $hack_class_to_graphql_interface,
-                ),
-            ];
-
-            if (!C\is_empty($mutation_fields)) {
-                $top_level_objects[] = new ObjectBuilder(
-                    new \Slack\GraphQL\__Private\CompositeType('Mutation', 'Mutation'),
-                    'Slack\\GraphQL\\Root',
-                    vec(Dict\sort_by_key($mutation_fields)),
-                    $hack_class_to_graphql_interface,
-                );
-            }
+        $objects[] = new ObjectBuilder(
+            new \Slack\GraphQL\__Private\CompositeType('Query', 'Query'),
+            'Slack\\GraphQL\\Root',
+            vec(Dict\sort_by_key($query_fields)),
+            $hack_class_to_graphql_interface,
+        );
 
-            $objects = Vec\concat($top_level_objects, $objects, $inputs);
+        if (!C\is_empty($mutation_fields)) {
+            $objects[] = new ObjectBuilder(
+                new \Slack\GraphQL\__Private\CompositeType('Mutation', 'Mutation'),
+                'Slack\\GraphQL\\Root',
+                vec(Dict\sort_by_key($mutation_fields)),
+                $hack_class_to_graphql_interface,
+            );
         }
 
-        return Vec\sort_by($objects, $object ==> $object->getGraphQLType());
+        return Vec\concat($objects, $inputs) |> Vec\sort_by($$, $object ==> $object->getGraphQLType());
```

Comment:
Why do we need a separate `inputs` vec? Can we just stick them in `$objects` too?

Reply:
Yea, this looks like it was just left over from something. Updated!